File: Saber/app/src/main/java/com/pathfinder/saber/About.java
package com.pathfinder.saber;

import android.content.Intent;
import android.net.Uri;
import android.provider.ContactsContract;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.Button;
import android.widget.ImageButton;

public class About extends AppCompatActivity {
    Window window;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_about);
        getWindow().getDecorView().setBackgroundColor(getColor(R.color.colorPrimaryDark));
        ImageButton close = findViewById(R.id.closeButt);
        window = this.getWindow();
        window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
        window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
        window.setStatusBarColor(this.getResources().getColor(R.color.colorPrimaryDark));
        window.setNavigationBarColor(this.getResources().getColor(R.color.colorPrimaryDark));
        close.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
                overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
            }
        });

        ImageButton insocial = findViewById(R.id.instaButt);
        insocial.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Uri uri = Uri.parse("https://www.instagram.com/ranjanistic");
                Intent web = new Intent(Intent.ACTION_VIEW, uri);
                startActivity(web);
            }
        });

        ImageButton twsocial = findViewById(R.id.tweetButt);
        twsocial.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Uri uri = Uri.parse("https://www.twitter.com/ranjanistic");
                Intent web = new Intent(Intent.ACTION_VIEW, uri);
                startActivity(web);
            }
        });
    }
}

File: Saber/app/src/main/java/com/pathfinder/saber/About.kt
package com.pathfinder.saber

import android.content.Intent
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.view.WindowManager
import android.widget.Button
import android.widget.ImageButton
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat

class About : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_about)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
        }
        window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            window.statusBarColor = ContextCompat.getColor(this,R.color.colorPrimaryDark)
            window.navigationBarColor = ContextCompat.getColor(this,R.color.colorPrimaryDark)
        }
        findViewById<ImageButton>(R.id.closeButt).setOnClickListener {
            finish()
            overridePendingTransition(R.anim.fadeliton, R.anim.fadelitoff)
        }
        findViewById<TextView>(R.id.version).text = BuildConfig.VERSION_NAME
        findViewById<Button>(R.id.premiumButton).setOnClickListener{
            startActivity(
                    Intent(Intent.ACTION_VIEW, Uri.parse("http://play.google.com/store/"))
            )
            //TODO: replace with saber+ link
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        overridePendingTransition(R.anim.fadeliton, R.anim.fadelitoff)
    }
}